port/threads.c

```diff
@@ -96,9 +95,9 @@ static void thread_waittid_thr(void *arg)
 			free(data->stack);
 			free(data);
 
-			mutexLock(global.join_lock);
-			condSignal(global.join_cond);
-			mutexUnlock(global.join_lock);
+			mutexLock(global.lock);
+			condBroadcast(global.join_cond);
+			mutexUnlock(global.lock);
```

Comment:
condSignal, condBroadcast does CPU reschedule, it's way better to call these without mutex.

condBroadcast -> reschedule -> mutexLock -> reschedule -> mutexUnlock -> reschedule
vs
condBroadcast -> reschedule -> mutexLock

Reply:
Both signal and broadcast execute this function: https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/8ee6add2b00884ea372506e542c74c53efc6a4f1/proc/threads.c#L1211
which does the reschedule if there was a thread on the queue